src/RA_Core.cpp

```diff
@@ -56,9 +56,11 @@ bool g_bRAMTamperedWith = false;
 BOOL APIENTRY DllMain(HMODULE hModule, DWORD dwReason, _UNUSED LPVOID)
 {
     if (dwReason == DLL_PROCESS_ATTACH)
+    {
         g_hThisDLLInst = hModule;
 
-    ra::services::Initialization::RegisterCoreServices();
+        ra::services::Initialization::RegisterCoreServices();
+    }
```

Comment:
This change is not directly related to the issue. It's just a minor optimization, as the services only need to be registered once on DLL load. When `DllMain` is called for `DLL_THREAD_ATTACH` or `DLL_PROCESS_DETACH`, this call does nothing as internally, it short-circuits if it's already been called.